Add Annotations to Action Runs

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR adds annotations to Actions runs to clearly highlight important issues from that run.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

These warnings indicate issues that prevent this project from updating a repository (or running at all) and it would help if they were displayed prominently to help them get noticed.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
Automated tests pass. You can see a [sample run](https://github.com/cisagov/action-lineage/actions/runs/1359832089) that has a number of the added annotations.
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Add Actions notations to important logging output

This adds annotations that will show up in the summary for the Action's run to
highlight important issues at a glance.

mcdonnnj committed:
Add error annotations for sanity checks

mcdonnnj committed:
Add warning annotation for incorrect lineage

This is another warning that would be best to bubble up into an annotation to
make it easier to notice the problem.

mcdonnnj committed:
Use the appropriate repository element

Use the `full_name` element of a repository's information instead of manually
creating it using `owner.login` and `repo.name`.

jsf9k commented on src/lineage/entrypoint.py:
```diff
@@ -292,22 +296,32 @@ def main() -> None:
 
     if repo_query is None:
         logging.fatal(
             "Reository query environment variable must be set. (INPUT_REPO_QUERY)"
```
```suggestion
            "Repository query environment variable must be set. (INPUT_REPO_QUERY)"
```

jsf9k commented on src/lineage/entrypoint.py:
```diff
@@ -203,8 +204,11 @@ def push(
     if not repo.permissions.push:
         logging.warning(
             "⚠️ WARNING! Missing 'push' permission on '%s/%s'",
```
You have two `%s` in the format string but only one argument.
```suggestion
            "⚠️ WARNING! Missing 'push' permission on '%s'",
```

jsf9k reviewed (approved):
I love it!  I did notice two typos.

mcdonnnj committed:
Fix typo in logging output

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Fix format string for logging output

The string needed to be updated to reflect the change in the number of inputs.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

dav3r reviewed (approved):
Noice! 🥇
